test/e2e/singlecluster/statefulset_test.go

```diff
@@ -325,5 +328,49 @@ var _ = ginkgo.Describe("StatefulSet integration", func() {
 				}, util.LongTimeout, util.Interval).Should(gomega.Succeed())
 			})
 		})
+
+		ginkgo.It("should allow to change queue name if ReadyReplicas=0", func() {
+			statefulSet := statefulsettesting.MakeStatefulSet("sts", ns.Name).
+				Image(util.E2eTestSleepImage, []string{"10m"}).
+				Request(corev1.ResourceCPU, "100m").
+				Replicas(3).
+				Queue(fmt.Sprintf("%s-invalid", localQueueName)).
```

Comment:
I think the more practical case for updating is setting if by users who forgot to set it in the first place.
So, I would suggest to remove it and set to the correct value. We could have both e2e tests, but maybe this is overkill.

Reply:
If a user forgets to set the queue name, the Pods should run. Is it valid case?

Reply:
Hmmm, I was thinking about scenario when using `manageJobsWithoutQueueName: true`. 
However, since our testing assumes `manageJobsWithoutQueueName: false` then the test case is correct. 
We can extend the testing after https://github.com/kubernetes-sigs/kueue/issues/3767, cc @kaisoz .

So feel free to keep as is.

Reply:
In case the Pods are already running, and we have the `manageJobsWithoutQueueName: false`, ideally we stop the STS and restart as managed by Kueue, but I'm not sure what is the priority, because no user raised this with us yet. They can always in all cases just re-create it. So, I'm not worried about the case yet. 

However, if you test that it does not work as expected, then it is better to prevent it by validation.

Reply:
We discussed it offline with @mimowo and decided to keep it as is for now.